ls1intum/Artemis
# `Development`: Move lectures' create, read, update and delete features into the lecture micro service

### Checklist
#### General


- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server (artemistest6).
#### Server
#### Client

### Motivation and Context


This PR is the last that moved features from Artemis to the lecture microservice. It moves features related to CRUD actions on lectures. 

### Description

**Note for code review:** LectureService, LectureResource and LectureIntegrationTest are copied from Artemis. They are not implemented by me and I think that their review can be skipped, but if you wish you can review them as well. Excluding those files reduced the changes with ~750 lines.
The client changes are related to the changed app name coming from the Lecture microservice. Since its name is lectureApp, messages starting with artemisApp.lecture.* had to be changed to lectureApp.lecture.*.

### Steps for Testing

Prerequisites:
- Test server 6: [https://artemistest6.ase.in.tum.de/](https://artemistest6.ase.in.tum.de/) - it is already deployed there
- 1 Instructor (artemis_test_user_12 or you can create another instructor account using the server administration)
- 1 Student (artemis_test_user_1 or you can create another student account using the server administration)

1. Log in to Artemis
2. Navigate to Course Management
3. Open/Create a course (the course should be deleted in the last steps) and create a lecture => lecture should be successfully created
5. Edit the lecture => the lecture should be successfully updated
6. Add attachments and 1-2 lecture units to the lecture (1 of them should be an exercise) => should work fine
7. Navigate to course overview and the lecture you created => the lecture should be there including the units and the attachments you added
8. Navigate to Course Management and delete the exercise that was linked to the lecture
9. Open the lecture and check that the exercise is removed from the lecture units list
10. Delete a lecture => the lecture should be successfully deleted

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots
